content/style-injector.js

```diff
@@ -94,40 +106,56 @@ function createStyleInjector({compare, setStyleContent, onUpdate}) {
     style.el.remove();
   }
 
-  function update({id, code}) {
+  function _update({id, code}) {
     const style = table.get(id);
     if (style.code === code) return;
     style.code = code;
     // workaround for Chrome devtools bug fixed in v65
-    // https://github.com/openstyles/stylus/commit/0fa391732ba8e35fa68f326a560fc04c04b8608b
-    let oldEl;
-    if (CHROME < 3321) {
-      oldEl = style.el;
+    if (isChromePre65) {
+      const oldEl = style.el;
       oldEl.id = '';
-      style.el = createStyle(id);
+      style.el = createStyle(id, code);
       oldEl.parentNode.insertBefore(style.el, oldEl.nextSibling);
-      style.el.disabled = !enabled;
+      oldEl.remove();
+    } else {
+      style.el.textContent = code;
+      style.el.sheet.disabled = !enabled;
+    }
+  }
+
+  function _supersede(domId) {
+    const el = document.getElementById(domId);
+    if (el) {
+      // remove if it looks like our style that wasn't cleaned up in orphanCheck
+      // (note, Firefox doesn't orphanize content scripts at all so orphanCheck will never run)
```

Comment:
orphanCheck doesn't receive the event we send so apparently Firefox automatically terminates the old content script's context.

Reply:
Then it might be better to remove all `#stylus-*` styles when `apply.js` is injected
```diff
diff --git a/content/apply.js b/content/apply.js
index 7b35b1fc..40ba1c1e 100644
--- a/content/apply.js
+++ b/content/apply.js
@@ -39,6 +39,13 @@ const APPLY = (() => {
     window.addEventListener(chrome.runtime.id, orphanCheck, true);
   }
 
+  if (FIREFOX) {
+    // firefox doesn't remove old elements
+    for (const el of document.querySelectorAll(`:root > style[id^=stylus-]`)) {
+      el.remove();
+    }
+  }
+
   let parentDomain;
 
   prefs.subscribe(['disableAll'], (key, value) => doDisableAll(value));
```

Reply:
Yes, it makes sense but I want to keep the individual check as well in case the site (or another extension/userscript) somehow messed with the style elements for example changed the element's id to stylus-NNN.